test/mock-server.js

```diff
@@ -3,8 +3,8 @@ module.exports = function(req, res) {
     if (req.url === '/mock/200ok') {
         res.statusCode = 200
         res.end('')
-    } else if (req.url === '/mock/no-content') {
-        res.statusCode = 204
+    } else if (req.url.substr(0,16) === '/mock/no-content') {
+        res.statusCode = ~~(req.url.substring(17))
```

Comment:
Nice thing about `~~` compared with `parseInt` is that they return `0` and `NaN` respectively for invalid input. But this is a mockserver, so understandability seems more important than fault-tolerance. 

I merged this already, so fixing in master.